jax/numpy/lax_numpy.py

```diff
@@ -2555,1276 +2765,1415 @@ def wrapper(*args, **kwargs):
+    res = argmin(a, axis=axis)
+    return where(all(nan_mask, axis=axis), -1, res)
 
 
 # TODO(mattjj): redo this lowering with a call to variadic lax.reduce
 def _argminmax(name, op, a, axis):
-  if a.shape[axis] == 0:
-    raise ValueError("attempt to get {} of an empty sequence".format(name))
-  shape = [1] * a.ndim
-  shape[axis] = a.shape[axis]
-  idxs = lax.tie_in(a, arange(a.shape[axis])).reshape(shape)
-  maxval = iinfo(dtypes.canonicalize_dtype(idxs.dtype)).max
-  maxval = lax.tie_in(a, maxval)
-  mask_idxs = where(lax._eq_meet(a, op(a, axis, keepdims=True)), idxs, maxval)
-  return min(mask_idxs, axis)
+    if a.shape[axis] == 0:
+        raise ValueError("attempt to get {} of an empty sequence".format(name))
+    shape = [1] * a.ndim
+    shape[axis] = a.shape[axis]
+    idxs = lax.tie_in(a, arange(a.shape[axis])).reshape(shape)
+    maxval = iinfo(dtypes.canonicalize_dtype(idxs.dtype)).max
+    maxval = lax.tie_in(a, maxval)
+    mask_idxs = where(lax._eq_meet(a, op(a, axis, keepdims=True)), idxs, maxval)
+    return min(mask_idxs, axis)
 
 
 @_wraps(np.sort)
-def sort(a, axis=-1, kind='quicksort', order=None):
-  if kind != 'quicksort':
-    warnings.warn("'kind' argument to sort is ignored.")
-  if order is not None:
-    raise ValueError("'order' argument to sort is not supported.")
+def sort(a, axis=-1, kind="quicksort", order=None):
+    if kind != "quicksort":
+        warnings.warn("'kind' argument to sort is ignored.")
+    if order is not None:
+        raise ValueError("'order' argument to sort is not supported.")
 
-  if axis is None:
-    return lax.sort(a.ravel(), dimension=0)
-  else:
-    return lax.sort(a, dimension=_canonicalize_axis(axis, ndim(a)))
+    if axis is None:
+        return lax.sort(a.ravel(), dimension=0)
+    else:
+        return lax.sort(a, dimension=_canonicalize_axis(axis, ndim(a)))
 
 
 @_wraps(np.argsort)
-def argsort(a, axis=-1, kind='quicksort', order=None):
-  if kind != 'quicksort':
-    warnings.warn("'kind' argument to argsort is ignored.")
-  if order is not None:
-    raise ValueError("'order' argument to argsort is not supported.")
-
-  if axis is None:
-    return argsort(a.ravel(), 0)
-  else:
-    axis = _canonicalize_axis(axis, ndim(a))
-    iota = lax.broadcasted_iota(np.int64, shape(a), axis)
-    _, perm = lax.sort_key_val(a, iota, dimension=axis)
-    return perm
+def argsort(a, axis=-1, kind="quicksort", order=None):
+    if kind != "quicksort":
+        warnings.warn("'kind' argument to argsort is ignored.")
+    if order is not None:
+        raise ValueError("'order' argument to argsort is not supported.")
+
+    if axis is None:
+        return argsort(a.ravel(), 0)
+    else:
+        axis = _canonicalize_axis(axis, ndim(a))
+        iota = lax.broadcasted_iota(np.int64, shape(a), axis)
+        _, perm = lax.sort_key_val(a, iota, dimension=axis)
+        return perm
+
+
+@_wraps(np.sort_complex)
+def sort_complex(a, axis=-1, kind="quicksort", order=None):
```

Comment:
Actually, on reflection, as far as I can tell NumPy's API doesn't have any of the additional arguments here, only the array:
https://numpy.org/doc/stable/reference/generated/numpy.sort_complex.html

We should not deviate from NumPy without a good reason.

Reply:
Thank you for your comments, I appreciate it. I will fix the formatting, and I will contact the author of the other PR, hopefully we can join efforts on this. Thanks again!